pkg/plugin/step.go

```diff
@@ -80,12 +81,20 @@ func calculateStep(minInterval time.Duration, from, to time.Time, maxDataPoints
 		resolution = defaultResolution
 	}
 
-	calculatedInterval := time.Duration((to.UnixNano() - from.UnixNano()) / resolution)
-	if calculatedInterval < minInterval {
-		return minInterval
+	rangeValue := to.UnixNano() - from.UnixNano()
+
+	calculatedInterval := time.Duration(rangeValue / resolution)
+	if calculatedInterval > minInterval {
+		calculatedInterval = minInterval
+	}
```

Comment:
Why the check is now inverted?

Reply:
Rollback this logic. used round function